korim/src/commonTest/kotlin/com/soywiz/korim/format/QOITest.kt

```diff
@@ -120,4 +163,18 @@ class QOITest {
         assertSame(qoiOutBitmap, output)
         assertFalse(qoiOutBitmap.premultiplied)
     }
+
+    @Test
+    fun providingSmallPreAllocatedArrayResultsInError() = suspendTestNoBrowser {
+        val original = QOI.decode(resourcesVfs["dice.qoi"])
+        val preallocatedArray = UByteArrayInt(4)
+        val props = ImageEncodingProps()
+        props.preAllocatedArrayForQOI = preallocatedArray
+        assertFailsWith<IllegalArgumentException> {
+            QOI.encode(
+                original,
+                props
```

Comment:
To keep it clean and one-liner when using external properties, I typically use apply (so you dont need to create a local variable and choose a name):

```
ImageEncodingProps().apply {
         preAllocatedArrayForQOI = preallocatedArray
}
```

Reply:
Sometimes I add to those class an optional extra lambda or a new constructor with a lambda that automatically calls it, so it is even cleaner:

```
ImageEncodingProps {
         preAllocatedArrayForQOI = preallocatedArray
}
```

```
class ImageEncodingProps(…, init: (ImageEncodingProps.() -> Unit)? = null) {
init { init?.invoke() }
}